backend/src/main/java/com/trackpack/app/exceptions/ResourceNotFoundException.java

```diff
@@ -0,0 +1,12 @@
+package com.trackpack.app.exceptions;
+
+import org.springframework.http.HttpStatus;
+import org.springframework.web.bind.annotation.ResponseStatus;
+
+@ResponseStatus(value = HttpStatus.NOT_FOUND)
+public class ResourceNotFoundException extends Exception{
```

Comment:
You made this exception as checked. I think this just make things more complicated, which we want to avoid. I'll make sense, if we can handle this Exception and get alternative data. But in fact (as I imagine it) we're just sending error message to client. And in code, we're just handling it at higher level. I think it will be more clearer if we just add general error-handling mechanism for spring controllers. And handle this exception in one method.